### ISSUE TITLE
finite difference of Hessian?

### ISSUE THREAD

**Issue by li-yang23** [vsingh-group/LCODEC-deep-unlearning#2]
I found in `scrub/grad_utils.py` that the Hessian is calculated using finite defferences like this:

```python
if approxType == 'FD':
        w1 = w1.to(hessian_device)
        w2 = w2.to(hessian_device)
    
        #hessian = torch.matmul((dw1 - dw2), (dw1 - dw2).transpose())
        grad_diff_outer = torch.einsum('p,q->pq', (dw1-dw2), (dw1-dw2))
 
        # divide by weight diff:
        pdist = torch.nn.PairwiseDistance(p=1)
        weight_scaling = pdist(w1.view(1,-1), w2.view(1,-1))

        hessian = torch.div(grad_diff_outer, weight_scaling)
```
I understand that this code divides the outer product of the gradient differences of the parameters in the last two epochs by the L1 norm of the parameter differences, but I don't understand why it is calculated this way.
I'm interested to know the theoretical basis for this calculation, because I tried to derive the formula but failed.

Is there any paper or book that I can find the reason to approximate Hessian in this way? Thank you.
